editor/export/editor_export_platform.cpp

```diff
@@ -831,29 +831,34 @@ String EditorExportPlatform::_get_script_encryption_key(const Ref<EditorExportPr
 	return p_preset->get_script_encryption_key().to_lower();
 }
 
-Vector<String> EditorExportPlatform::get_forced_export_files() {
+Vector<String> EditorExportPlatform::get_forced_export_files(bool p_main_pack) {
 	Vector<String> files;
 
-	files.push_back(ProjectSettings::get_singleton()->get_global_class_list_path());
+	// these files are only needed if we are exporting a main pack file
+	if (p_main_pack) {
+		String icon = GLOBAL_GET("application/config/icon");
+		if (!icon.is_empty() && FileAccess::exists(icon)) {
+			files.push_back(icon);
+		}
 
-	String icon = GLOBAL_GET("application/config/icon");
-	String splash = GLOBAL_GET("application/boot_splash/image");
-	if (!icon.is_empty() && FileAccess::exists(icon)) {
-		files.push_back(icon);
-	}
-	if (!splash.is_empty() && FileAccess::exists(splash) && icon != splash) {
-		files.push_back(splash);
+		String splash = GLOBAL_GET("application/boot_splash/image");
+		if (!splash.is_empty() && icon != splash && FileAccess::exists(splash)) {
+			files.push_back(splash);
+		}
+
+		String extension_list_config_file = GDExtension::get_extension_list_config_file();
+		if (FileAccess::exists(extension_list_config_file)) {
+			files.push_back(extension_list_config_file);
+		}
 	}
+
+	files.push_back(ProjectSettings::get_singleton()->get_global_class_list_path());
+
 	String resource_cache_file = ResourceUID::get_cache_file();
 	if (FileAccess::exists(resource_cache_file)) {
 		files.push_back(resource_cache_file);
 	}
 
-	String extension_list_config_file = GDExtension::get_extension_list_config_file();
-	if (FileAccess::exists(extension_list_config_file)) {
-		files.push_back(extension_list_config_file);
-	}
-
 	// Store text server data if it is supported.
 	if (TS->has_feature(TextServer::FEATURE_USE_SUPPORT_DATA)) {
 		bool use_data = GLOBAL_GET("internationalization/locale/include_text_server_data");
```

Comment:
Should this be in each pck too, or only the main pack?

If only the main pack, this is starting to look like it should just be two separate functions as there's little overlap between the two scenarios.

Reply:
To be honest, I wasn't sure how this worked. Conceptually, if this is a translation DB, then it probably needs to be in all PCKs since they could potentially introduce new strings. That said, it probably would need a similar merge process as the UID cache and global script class cache to work properly. I'm not really comfortable enough with the localization system to attempt that fix. 

In lieu of that, I can either pack it in the main pck only (trimming something that's possibly unused) or keep it in all pcks (preserving current behavior). Thoughts? Happy to split this function in two if the call is to main-pck it.

Reply:
I believe it's a dictionary that's not specific to the game translations / pck strings, but @bruvzg should be able to confirm.